src/Controllers/Controller.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace Masterclass\Controllers;
+
+
+use PDO;
+
+/**
+ * Class Controller
+ * @package Masterclass\Controllers
+ */
+class Controller
```

Comment:
Here be dragons.

Creating a base controller can seem like it's going to save you a lot of work, but it often doesn't. You're defining a constructor that you're going to have to live with forever, meanwhile you may have objects that you don't need/want that are always required.